Userland/Libraries/LibGUI/PathBreadcrumbbar.cpp

```diff
@@ -0,0 +1,178 @@
+/*
+ * Copyright (c) 2018-2021, Andreas Kling <kling@serenityos.org>
+ * Copyright (c) 2021-2023, Sam Atkins <atkinssj@serenityos.org>
+ * Copyright (c) 2021, Mustafa Quraish <mustafa@cs.toronto.edu>
+ * Copyright (c) 2022, the SerenityOS developers.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include "PathBreadcrumbbar.h"
+#include <AK/LexicalPath.h>
+#include <LibCore/File.h>
+#include <LibCore/MimeData.h>
+#include <LibGUI/BoxLayout.h>
+#include <LibGUI/Breadcrumbbar.h>
+#include <LibGUI/FileIconProvider.h>
+#include <LibGUI/Icon.h>
+#include <LibGUI/TextBox.h>
+
+REGISTER_WIDGET(GUI, PathBreadcrumbbar)
+
+namespace GUI {
+
+ErrorOr<NonnullRefPtr<PathBreadcrumbbar>> PathBreadcrumbbar::try_create()
+{
+    auto location_text_box = TRY(TextBox::try_create());
+    auto breadcrumbbar = TRY(Breadcrumbbar::try_create());
+
+    auto path_breadcrumbbar = TRY(adopt_nonnull_ref_or_enomem(new (nothrow) PathBreadcrumbbar(*location_text_box, *breadcrumbbar)));
+    (void)TRY(path_breadcrumbbar->try_set_layout<GUI::VerticalBoxLayout>());
+    TRY(path_breadcrumbbar->try_add_child(location_text_box));
+    TRY(path_breadcrumbbar->try_add_child(breadcrumbbar));
+
+    return path_breadcrumbbar;
+}
```

Comment:
From the recent discussions about initializing GUI widgets, here's what I came up with. Creating all the child widgets separately, then passing them in to the constructor so its fields can be NonnullRefPtrs.

Loading from a GML string would be... interesting though.